### PR TITLE
feat: update Elixir SDK APIs

### PR DESCRIPTION
**Pull request by wingyplus**
This changes update engine version to 0.8.0 including re-generate APIs to be the same with engine dev.

### CODE REVIEW

File: sdk/elixir/lib/dagger/engine_conn.ex [hunk 1]
```diff
@@ -5,7 +5,7 @@ defmodule Dagger.EngineConn do
 
   defstruct [:port, :token, :session_pid]
 
-  @dagger_cli_version "0.6.2"
+  @dagger_cli_version "0.6.4"
```
Comment (gerhard) [inline sdk/elixir/lib/dagger/engine_conn.ex:8]: Was this meant to be `0.8.0`?

BTW, `0.8.1` is coming in hot, most likely in the next few hours: https://github.com/dagger/dagger/milestone/19

File: sdk/rust/crates/dagger-sdk/src/core/mod.rs [hunk 2]
```diff
@@ -1,4 +1,4 @@
-pub const DAGGER_ENGINE_VERSION: &'static str = "0.6.4";
+pub const DAGGER_ENGINE_VERSION: &'static str = "0.8.0";
```
Comment (gerhard) [inline sdk/rust/crates/dagger-sdk/src/core/mod.rs:1]: I don't think that this change is meant to be in this PR.

Comment (wingyplus) [reply sdk/rust/crates/dagger-sdk/src/core/mod.rs:1]: I’m running the wrong bump. T_T Let me revert the change.

File: sdk/elixir/lib/dagger/engine_conn.ex (hunk 1 above)
Comment (wingyplus) [reply sdk/elixir/lib/dagger/engine_conn.ex:8]: I’ll re-bump to 0.8.1. 🎉

File: sdk/rust/crates/dagger-sdk/src/core/mod.rs (hunk 2 above)
Comment (wingyplus) [reply sdk/rust/crates/dagger-sdk/src/core/mod.rs]: Fix this issue in https://github.com/dagger/dagger/pull/5565/commits/7523310d8cb6a777aace0a283e1d1491a09f1942

File: sdk/elixir/lib/dagger/engine_conn.ex (hunk 1 above)
Comment (wingyplus) [reply sdk/elixir/lib/dagger/engine_conn.ex]: Fix in this issue https://github.com/dagger/dagger/pull/5565/commits/7523310d8cb6a777aace0a283e1d1491a09f1942

**Review by gerhard — APPROVED** [review]
👍 🚀

**Pull request merged**